homeassistant/components/here_weather/sensor.py

```diff
@@ -0,0 +1,99 @@
+"""Support for the HERE Destination Weather service."""
+import logging
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_MODE, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import HEREWeatherData
+from .const import DOMAIN, SENSOR_TYPES
+from .utils import convert_unit_of_measurement_if_needed, get_attribute_from_here_data
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: ConfigEntry, async_add_entities
+):
+    """Add here_weather entities from a config_entry."""
+    here_weather_data = hass.data[DOMAIN][config_entry.entry_id]
+
+    sensors_to_add = []
+    for sensor_type in SENSOR_TYPES:
+        if sensor_type == config_entry.data[CONF_MODE]:
```

Comment:
Don't use a config variable to enable/disable entities. Instead leverage `entity_registry_enabled_default` to disable things by default.

Reply:
Sorry I don't understand what you mean. Given what mode the user chooses there are completely different sensor entities. For example if the user chooses Astronomy he will get a sunrise sensor.

Could you give me an example integration which implements this in the way you mean?

Reply:
Shelly: https://github.com/home-assistant/core/blob/dev/homeassistant/components/shelly/binary_sensor.py#L56-L59

We should try to avoid allowing users to manage entities in config flows, since that duplicates the logic that exists in the entity registry.

Reply:
The thing here is that the config flow determines [here](https://github.com/home-assistant/core/blob/87415ef7bbb19c5bc16308b8a68ac1ffa246f64f/homeassistant/components/here_weather/__init__.py#L78) which api call should be made [in the update method](https://github.com/home-assistant/core/blob/87415ef7bbb19c5bc16308b8a68ac1ffa246f64f/homeassistant/components/here_weather/__init__.py#L127).

Based on this setting either entities defined in `ASTRONOMY_ATTRIBUTES `, **or** in `HOURLY_ATTRIBUTES ` **or** .... are present in the answer from the api call.
This means it does indeed makes sense to disable by default some of the sensors of the e.g. `ASTRONOMY_ATTRIBUTES `. But if another mode is chosen in the config flow we will never have the `sunrise` sensor in the api response.

Reply:
So this would require a data update coordinator per mode. Each entity will subscribe to the data update coordinator that has the data they are interested in .

Reply:
Okay, thank you!
This means I would than have to implement a logic in the update coordinators which checks for the configured mode(s) (configured via an options flow) in order to avoid multiple api calls -> Don't request astronomy data, hourly data,.... and thus waste free api calls if the user is only interested in daily data.
Do you concur?

Reply:
If you have different entity update coordinators per mode, an entity will only subscribe to the update coordinator that it wants data from. Update coordinators will not update their data if no one is subscribed.

Reply:
Done